Samples/Tutorial Samples/WinMLPIXSample/WinMLPIXSample/main.cpp

```diff
@@ -0,0 +1,129 @@
+﻿#include "pch.h"
+#include "PixRedistMiddleware.h"
+#include "WinMLHelper.h"
+
+using namespace winrt;
+using namespace Microsoft::WRL;
+using namespace Microsoft::WRL::Wrappers;
+using namespace std;
+
+#ifndef USE_PIX
+#define USE_PIX
+#endif
+
+static hstring pixCaptureFilePath = L".\\PIXCapture\\capture.wpix";
+
+ID3D12Device* m_d3d12Device;
+ID3D12CommandList* m_commandList;
+ID3D12CommandQueue* m_commandQueue;
+UINT m_color = PIX_COLOR(255, 0, 0);
+
+void GetHardwareAdapter(IDXGIFactory4* pFactory, IDXGIAdapter1** ppAdapter)
+{
+    *ppAdapter = nullptr;
+    for (UINT adapterIndex = 0; ; ++adapterIndex)
+    {
+        IDXGIAdapter1* pAdapter = nullptr;
+        if (DXGI_ERROR_NOT_FOUND == pFactory->EnumAdapters1(adapterIndex, &pAdapter))
+        {
+            // No more adapters to enumerate.
+            break;
+        }
+
+        // Check to see if the adapter supports Direct3D 12, but don't create the
+        // actual device yet.
+        if (SUCCEEDED(D3D12CreateDevice(pAdapter, D3D_FEATURE_LEVEL_12_0, _uuidof(ID3D12Device), nullptr)))
+        {
+            *ppAdapter = pAdapter;
+            return;
+        }
+        pAdapter->Release();
+    }
+}
+
+PIXCaptureParameters GetCap()
+{
+    PIXCaptureParameters cap;
+    cap.GpuCaptureParameters.FileName = pixCaptureFilePath.c_str();
+    return cap;
+}
+
+// Create command queue & command list to be used by WinML
+void GetD3D12CommandAssets()
+{
+    D3D12_COMMAND_LIST_TYPE type = D3D12_COMMAND_LIST_TYPE_DIRECT;
+    D3D12_COMMAND_QUEUE_DESC queueDesc = {};
+    queueDesc.Type = type;
+    queueDesc.Flags = D3D12_COMMAND_QUEUE_FLAG_NONE;
+    ID3D12CommandAllocator* commandAllocator;
+
+    IDXGIFactory4* factory;
+    CreateDXGIFactory1(IID_PPV_ARGS(&factory));
+
+    IDXGIAdapter1* hardwareAdapter;
+    GetHardwareAdapter(factory, &hardwareAdapter);
+
+    D3D12CreateDevice(hardwareAdapter, D3D_FEATURE_LEVEL_12_0, _uuidof(ID3D12Device), (void**)&m_d3d12Device);
+
+    m_d3d12Device->CreateCommandAllocator(
+        type,
+        IID_PPV_ARGS(&commandAllocator)
+    );
+    m_d3d12Device->CreateCommandQueue(&queueDesc, IID_PPV_ARGS(&m_commandQueue));
+    m_d3d12Device->CreateCommandList(0, type, commandAllocator, nullptr, __uuidof(ID3D12CommandList), (void**)&m_commandList);
```

Comment:
can you use a constant for the 0 to explain what it's for?

Reply:
There doesn't seem to be system defined constants defined for node mask, the example shown in this page uses value 0:
https://docs.microsoft.com/en-us/windows/win32/api/d3d12/nf-d3d12-id3d12device-createcommandlist

Reply:
yeah, looks like the standard is to just use 0. ok.